airbyte-commons-worker/src/main/java/io/airbyte/workers/helper/PersistConfigHelper.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.helper;
+
+import com.google.common.hash.Hashing;
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.generated.DestinationApi;
+import io.airbyte.api.client.generated.JobsApi;
+import io.airbyte.api.client.generated.SourceApi;
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.ConnectionRead;
+import io.airbyte.api.client.model.generated.DestinationIdRequestBody;
+import io.airbyte.api.client.model.generated.DestinationRead;
+import io.airbyte.api.client.model.generated.DestinationUpdate;
+import io.airbyte.api.client.model.generated.JobIdRequestBody;
+import io.airbyte.api.client.model.generated.JobInfoLightRead;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.api.client.model.generated.SourceUpdate;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.Config;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for workers to persist updates to Source/Destination configs emitted from
+ * AirbyteControlMessages.
+ */
+public class PersistConfigHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PersistConfigHelper.class);
+
+  private final AirbyteApiClient apiClient;
+
+  public PersistConfigHelper(final AirbyteApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  private UUID getConnectionIdFromJobId(final Long jobId) throws ApiException {
+    final JobsApi jobsApi = apiClient.getJobsApi();
+    final JobIdRequestBody body = new JobIdRequestBody().id(jobId);
+    final JobInfoLightRead jobInfo = jobsApi.getJobInfoLight(body);
+    return UUID.fromString(jobInfo.getJob().getConfigId());
+  }
+
+  public void persistSourceConfig(final Long jobId, final Config config) throws ApiException {
+    final UUID connectionId = getConnectionIdFromJobId(jobId);
+
+    final ConnectionRead connection = apiClient.getConnectionApi().getConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    final UUID sourceId = connection.getSourceId();
+
+    final SourceApi sourceApi = apiClient.getSourceApi();
+    final SourceRead source = sourceApi.getSource(new SourceIdRequestBody().sourceId(sourceId));
+
+    final SourceRead updatedSource = sourceApi
+        .updateSource(new SourceUpdate()
+            .sourceId(sourceId)
+            .name(source.getName())
+            .connectionConfiguration(Jsons.jsonNode(config.getAdditionalProperties())));
```

Comment:
I don't see anything in here about authentication... I assume that's handled for us?

Reply:
The ApiClientBeanFactory used by micronaut to initialize the ApiClient takes care of setting up the client with any auth-related / evironment-dependent functionality:

https://github.com/airbytehq/airbyte/blob/c30e8f64002d4042121b3b9b6284d75cfc054ea1/airbyte-commons-worker/src/main/java/io/airbyte/workers/config/ApiClientBeanFactory.java#L33

Reply:
☕